### PR TITLE
ctrl + backspace support on chat (Fixes ddnet#515)

### PR DESCRIPTION
**Pull request by Ryozuki**
- Type something, hit ctrl+backspace and it will delete the word (works like ctrl+left)

### CODE REVIEW

File: src/game/client/components/chat.cpp [hunk 1]
```diff
@@ -156,28 +156,41 @@ bool CChat::OnInput(IInput::CEvent Event)
...
+					if (SearchDirection < 0)
 						FoundAt++;
 					break;
 				}
 			}
+			if(Input()->KeyPress(KEY_BACKSPACE))
+			{
+				if(m_Input.GetCursorOffset() != 0)
+				{
+					char Text[512];
```
Comment (heinrich5991) [inline src/game/client/components/chat.cpp]: Should be `aText` because it is an array.

File: src/game/client/components/chat.cpp [hunk 2]
```diff
@@ -156,28 +156,41 @@ bool CChat::OnInput(IInput::CEvent Event)
...
 
 		if(SearchDirection != 0)
 		{
-			int FoundAt = SearchDirection > 0 ? m_Input.GetLength()-1 : 0;
-			for(int i = m_Input.GetCursorOffset()+SearchDirection; SearchDirection > 0 ? i < m_Input.GetLength()-1 : i > 0; i+=SearchDirection)
+			int FoundAt = SearchDirection > 0 ? m_Input.GetLength() - 1 : 0;
+			for(int i = m_Input.GetCursorOffset() + SearchDirection; SearchDirection > 0 ? i < m_Input.GetLength() - 1 : i > 0; i += SearchDirection)
 			{
-				int next = i+SearchDirection;
+				int next = i + SearchDirection;
```
Comment (heinrich5991) [inline src/game/client/components/chat.cpp]: While you're at it, this should be `Next` (starting with a capital letter).

**Comment by Ryozuki** [comment]
Done :)

**Comment by heinrich5991** [comment]
Thanks again!

**Pull request merged**